xilinx/xdc.cc

```diff
@@ -115,69 +121,73 @@ void Arch::parseXdc(std::istream &in)
 
     while (std::getline(in, line)) {
         ++lineno;
-        // Trim comments, from # until end of the line
-        size_t cstart = line.find('#');
-        if (cstart != std::string::npos)
-            line = line.substr(0, cstart);
-        if (isempty(line))
-            continue;
-
-        std::vector<std::string> arguments = split_to_args(line, true);
-        if (arguments.empty())
-            continue;
-        std::string &cmd = arguments.front();
-        if (cmd == "set_property") {
-            std::vector<std::pair<std::string, std::string>> arg_pairs;
-            if (arguments.size() != 4)
-                log_error("expected four arguments to 'set_property' (on line %d)\n", lineno);
-            else if (arguments.at(1) == "-dict") {
-                std::vector<std::string> dict_args = split_to_args(strip_quotes(arguments.at(2)), false);
-                if ((dict_args.size() % 2) != 0)
-                    log_error("expected an even number of argument for dictionary (on line %d)\n", lineno);
-                arg_pairs.reserve(dict_args.size() / 2);
-                for (int cursor = 0; cursor + 1 < int(dict_args.size()); cursor += 2) {
-                    arg_pairs.emplace_back(std::move(dict_args.at(cursor)), std::move(dict_args.at(cursor + 1)));
-                }
-            } else
-                arg_pairs.emplace_back(std::move(arguments.at(1)), std::move(arguments.at(2)));
-            if (arguments.at(1) == "INTERNAL_VREF")
+        try{ 
+            // Trim comments, from # until end of the line
```

Comment:
I can make any change you prefer here if you would like to help preserve line/blame history. Everything inside this `try` is the same as before.

Reply:
Not necessary, thanks.